server/bin/pbench-verify-indexed.py

```diff
@@ -61,105 +51,114 @@ def gen_tb_list(archive):
-                        # tar ball pattern.  Such entries should be flagged by
-                        # the server audit process.
+                        # tar ball pattern.
                         continue
                     yield c_entry.name, entry.name
 
 
 def main(options):
-    if not options.cfg_name:
-        print(
-            f"{_NAME_}: ERROR: No config file specified; set"
-            " _PBENCH_SERVER_CONFIG env variable",
-            file=sys.stderr,
-        )
-        return 1
+    """Simple driver function for this CLI.
 
+    Prints a report summarizing what was found, and then provides detailed
+    lists of properly and improperly indexed tar balls.
+
+    Returns 0 on success, 1 on failure.
+    """
     try:
         config = pbench.PbenchConfig(options.cfg_name)
     except pbench.BadConfig as e:
         print(f"{_NAME_}: {e}", file=sys.stderr)
-        return 2
+        return 1
 
     try:
         archive_p = Path(config.ARCHIVE).resolve(strict=True)
     except FileNotFoundError:
         print(
-            f"The configured ARCHIVE directory, {config.ARCHIVE}, does not exist",
+            f"The configured ARCHIVE directory, {config.ARCHIVE}," " does not exist",
             file=sys.stderr,
         )
-        return 3
+        return 1
 
     if not archive_p.is_dir():
         print(
-            f"The configured ARCHIVE directory, {config.ARCHIVE}, is not a valid directory",
+            f"The configured ARCHIVE directory, {config.ARCHIVE},"
+            " is not a valid directory",
             file=sys.stderr,
         )
-        return 4
+        return 1
 
     print("Verifying indexed state of tar balls")
-    start = pbench._time()
 
-    es = Elasticsearch(
-        [
-            {
-                "host": "elasticsearch.perf.lab.eng.bos.redhat.com",
-                "port": "9280",
-                "timeout": 600,
-            },
-        ]
+    es_client = Elasticsearch(
+        pbench.indexer._get_es_hosts(config, logging.getLogger("pbench-verify-indexed"))
     )
-    body = {
-        "query": {
-            "filtered": {
-                "query": {"query_string": {"analyze_wildcard": True, "query": "*"}}
-            }
-        }
-    }
-    res = es.search(
-        index="dsa-pbench.v4.run.*",
+    query = {"query": {"match_all": {}}}
+
+    # Start the overall timer.
+    start = pbench._time()
+
+    index_prefix = config.get("Indexing", "index_prefix")
+    if not index_prefix:
+        print(
+            "Missing 'index_prefix' value in 'Indexing' section of configuration",
+            file=sys.stderr,
+        )
+        return 1
+
+    # Fetch a thousand records at a time.
+    scanner = helpers.scan(
+        es_client,
+        index=f"{index_prefix}.v4.run.*",
         doc_type="pbench-run",
+        query=query,
         expand_wildcards="open",
-        body=body,
         fields=["@metadata.file-name"],
-        size=100000,
+        size=1000,
```

Comment:
This is (still) a magic number which should be declared and referenced symbolically.  (And, I suspect that 1000 records is pretty small, depending on how big each record is.)

Reply:
Yes, definitely a magic number.  It is going to be a magic number whether or not it is symbolically referenced.  I have made that change, but it seems rather "empty" to do on the `b0.69` branch.

Reply:
The point is that it's a "configuration parameter" or a "tunable", etc.  Those should be declared at the top of the class or module, or in a definitions file, so that they can be easily found and modified, rather than having to dive into the code to make adjustments.  Moreover, it's nice to have the semantic context which a symbolic reference provides.

Yeah, given that it's coming late to the game in the `b0.69`, it's a less satisfying effort, but it doing it is good practice and the next time we're tempted to encode a magic number we'll be more likely to do it symbolically.